File: Java/ContaCorrente.java
public class ContaCorrente {
    private double saldo;

    public ContaCorrente(double saldoInicial) {
        this.saldo = saldoInicial;
    }

    public void deposito(double valor) {
        saldo += valor;
    }

    public void saque(double valor) {
        if (valor <= saldo) {
            saldo -= valor;
        } else {
            System.out.println("Saldo insuficiente!");
        }
    }

    public void transferencia(ContaCorrente contaDestino, double valor) {
        if (valor <= saldo) {
            saldo -= valor;
            contaDestino.deposito(valor);
        } else {
            System.out.println("Saldo insuficiente para a transferência!");
        }
    }

    public double getSaldo() {
        return saldo;
    }
}

File: Kotlin/ContaCorrente.kt
class ContaCorrente(var saldo: Double) {

    fun depositar(valor: Double) {
        saldo += valor
        println("Depósito de R$$valor realizado. Saldo atual: R$$saldo")
    }

    fun sacar(valor: Double) {
        if (saldo >= valor) {
            saldo -= valor
            println("Saque de R$$valor realizado. Saldo atual: R$$saldo")
        } else {
            println("Saldo insuficiente!")
        }
    }

    fun transferir(valor: Double, contaDestino: ContaCorrente) {
        if (saldo >= valor) {
            saldo -= valor
            contaDestino.depositar(valor)
            println("Transferência de R$$valor realizada. Saldo atual: R$$saldo")
        } else {
            println("Saldo insuficiente!")
        }
    }
}
